pkg/cloud/aws/services/ec2/instances.go

```diff
@@ -325,7 +325,7 @@ func (s *Service) createInstance(machine *actuators.MachineScope, bootstrapToken
 				kubeadm.SetNodeRegistrationOptions(
 					&machine.MachineConfig.KubeadmConfiguration.Join.NodeRegistration,
 					kubeadm.WithNodeRegistrationName(hostnameLookup),
-					kubeadm.WithCRISocket(containerdSocket),
+					kubeadm.WithDefaultCRISocket(containerdSocket),
```

Comment:
My suggestion would be to figure out which socket to use above this instantiation logic and then use the existing method to set the CRISocket

is that possible with your use case?